src/rest.js

```diff
@@ -14,53 +14,50 @@
  * limitations under the License.
  */
 
-var cachedResponseLastRefresh = null;
-var cachedResponse = null;
-
 function sql(datasetKey, sqlQuery, lastRefresh, sampleExtraction) {
-    if (!(lastRefresh && cachedResponseLastRefresh) ||
-        (lastRefresh > cachedResponseLastRefresh)) {
-
-        var params = {
-            method: 'post',
-            headers: {
-                'Accept': 'application/x-ndjson',
-                'Authorization': 'Bearer ' + getOAuthService().getAccessToken()
-            },
-            payload: {
-                'query': sqlQuery
-            },
-            muteHttpExceptions: true
-        };
+    var params = {
+        method: 'post',
+        headers: {
+            'Accept': 'application/x-ndjson',
+            'Authorization': 'Bearer ' + getOAuthService().getAccessToken()
+        },
+        payload: {
+            'query': sqlQuery
+        },
+        muteHttpExceptions: true
+    };
 
-        var response = UrlFetchApp.fetch(getSqlEndpoint(datasetKey), params);
-        if (response.getResponseCode() === 200) {
-            cachedResponse = response;
-            cachedResponseLastRefresh = lastRefresh;
+    var success = false;
+    var lock = LockService.getUserLock();
+    success = lock.tryLock(60000);
```

Comment:
why `60000` for timeout?

Reply:
Maximum query result size from data.world connector is 25MB. Rest API will take ~2 seconds to query the result with 25MB data. I  assumed that there will not be more than 30 REST API call simultaneously and there may not always 25 MB of data in the query result. .  (Number 30 is a wild guess ).

 if you feel timeout 60 seconds is less or logic needs to change, please let me know.

Reply:
I have increased the timeout to 3minutes